ast/edit/option_editor_test.go

```diff
@@ -0,0 +1,242 @@
+package edit_test
+
+import (
+	"testing"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/ast/edit"
+	"github.com/influxdata/flux/parser"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+	"github.com/pkg/errors"
+)
+
+func TestEditor(t *testing.T) {
+	testCases := []struct {
+		name         string
+		in           string
+		edited       []string // slice for multiple accepted results
+		editExpected bool
+		edit         func(node ast.Node) (bool, error)
+	}{
+		{
+			name: "no_option",
+			in: `from(bucket: "test")
+	|> range(start: 2018-05-23T13:09:22.885021542Z)`,
+			edited: []string{`from(bucket: "test")
+	|> range(start: 2018-05-23T13:09:22.885021542Z)`},
+			editExpected: false,
+			edit: func(node ast.Node) (bool, error) {
+				return edit.Edit(node, "from", nil)
+			},
+		},
+		{
+			name:         "option_wrong_id",
+			in:           `option foo = 1`,
+			edited:       []string{`option foo = 1`},
+			editExpected: false,
+			edit: func(node ast.Node) (bool, error) {
+				return edit.Edit(node, "bar", nil)
+			},
+		},
+		{
+			name: "sets_option",
+			in: `option foo = 1
+option bar = 1`,
+			edited: []string{`option foo = 1
+option bar = 42`},
+			editExpected: true,
+			edit: func(node ast.Node) (bool, error) {
+				literal := edit.GetLiteral(values.New(int64(42)))
+				return edit.Edit(node, "bar", edit.EditValueFn(literal))
+			},
+		},
+		{
+			name: "updates_object",
+			in: `option foo = 1
+option task = {
+	name: "bar",
+	every: 1m,
+	delay: 1m,
+	cron: "20 * * *",
+	retry: 5,
+}`,
+			edited: []string{`option foo = 1
+option task = {
+	name: "bar",
+	every: 7200000000000ns,
+	delay: 2520000000000ns,
+	cron: "buz",
+	retry: 10,
+}`},
+			editExpected: true,
+			edit: func(node ast.Node) (bool, error) {
+				return edit.Edit(node, "task", edit.EditObjectFn(map[string]values.Value{
+					"foo":   values.New("foo"), // should ignore this
+					"bar":   values.New("bar"), // should ignore this
+					"buz":   values.New("buz"), // should ignore this
+					"every": values.New(duration("2h")),
+					"delay": values.New(duration("42m")),
```

Comment:
This can be easily written as `42*time.Minute`, this way you do not have to create the helper function to parse it. And it still reads clearly.